AWS: Add unit tests for GlueCatalog's isValidIdentifier method

rajarshisarkar opened the pull request:
- I changed `GlueCatalog.isValidIdentifier` to return `false` always and then ran `TestGlueCatalog` (all 21 test cases passed).
- I changed `GlueCatalog.isValidIdentifier` to return `true` always and then ran `TestGlueCatalog` (all 21 test cases passed).

The current master doesn't seem to have unit tests to check the behaviour of `GlueCatalog.isValidIdentifier`. Hence, this PR adds few unit tests.

rdblue commented:
@jackye1995 can you review?

jackye1995 commented:
I was holding on this review because I thought it's related to the discussion in #3668 , but more test is always better. Where were you trying to make change? Could you describe in more details? Maybe we missed some test cases somewhere else.

rajarshisarkar commented:
Hi @jackye1995, This change is independent of https://github.com/apache/iceberg/pull/3668. 

- I changed `GlueCatalog.isValidIdentifier` to return `false` always and then ran `TestGlueCatalog` (all 21 test cases passed).
- I changed `GlueCatalog.isValidIdentifier` to return `true` always and then ran `TestGlueCatalog` (all 21 test cases passed).

The current master doesn't seem to have unit tests to check the behaviour of `GlueCatalog.isValidIdentifier`. Hence, this PR adds few unit tests.

jackye1995 commented:
I think that is already tested in `testCreateNamespaceBadName` and `testToDatabaseNameFailure`. `TestGlueCatalog` does not have the table name test because verification of the table name is done in `BaseMetastoreCatalog` through `isValidIdentifier`.

But I agree the current `TestGlueCatalog` lacks a lot of features. I am thinking about implementing a proper local Glue client to test the code in unit test properly, let me know if you would like to pick up the task.
